### PR TITLE
Added named function

### PR DESCRIPTION
**Pull request by beardedtim**
## What?

This PR adds a named function instead of having the anonymous function handle the question.

## Why?

This is the start of adding more logic and cleaning up.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,8 +7,17 @@ const rl = readline.createInterface({
 
 const items = ["pistol", "broadsword", "dagger", "health potion"];
 
-rl.question('What do you think of Node.js? ', (answer) => {
-  console.log(`Thank you for your valuable feedback: ${answer}`);
+const hasItem = item => items.find(i => i === item)
```
Comment (kylelee33) [inline index.js]: Should this be formatted like this? 
`const hasItem = (item) => {
items.find(i => i === item)
}`

Comment (beardedtim) [reply index.js]: This is a great question! With `arrow functions`, you can have what's called [`implicit return`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Function_body) which means we can say that these functions are equivalent

```js
const add = (a, b) => a + b

const add = function(a,b){
  return a + b
}

function add(a, b) {
  return a + b
}
```

With the code block in question, we want to _return_ the value of `items.find` as the result of calling the function `hasItem`. We can see that we check the "truthiness" of the return value in the `if` check

```js
if(hasItem(answer)) {

}
```

which could be written as

```js
if(items.find(item => item === answer)) {

}
```

Comment (beardedtim) [reply index.js]: To put it another way, when we say

```js
const hasItem = item => { items.find(...) }
```

_where_ is the value that `items.find` is creating? How does that get outside of the `{...}`? Where does that value get _assigned_?

Comment (beardedtim) [reply index.js]: When we write

```js
const hasItem = item => items.find(...)
```

we are saying we can replace `hasItem(answer)` with _the result of items.find(...)_ and them be the same thing.

Comment (kylelee33) [reply index.js]: So if hasItem is a function, you don't need the paranthesis around `item`? or is it something besides a function

Comment (beardedtim) [reply index.js]: > So if hasItem is a function, you don't need the paranthesis around item? 

Gah, yeah, that's probably confusing. My bad.

With arrow functions, when you `declare` them, there are a few rules:

1. If there is no arguments to the function, you **must** have the parentheses 
  - `const a = () => a`
2. If there is _**only**_ one argument, you **can** omit the parentheses 
  - `const a = b => b`
3. If there is more than one, you **must** have the parentheses 
  - `const a = (b, c) => d`

Comment (beardedtim) [reply index.js]: So these are equivalent

```js
const a = b => b

const a = function(b) {
  return b
}

function a(b) {
  return b
}
```

_**Note**_: if it is not an arrow function, you _must_ have the parentheses

Comment (kylelee33) [reply index.js]: OK that clears it up for me, thank you!

**Pull request merged**